src/test/ui/unsized-locals/unsized-exprs.rs

```diff
@@ -1,4 +1,5 @@
-#![feature(unsized_tuple_coercion, unsized_locals)]
+#![feature(unsized_tuple_coercion, unsized_locals, unsized_fn_params)]
```

Comment:
Why does this need `unsized_locals`?

Reply:
```
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> src/test/ui/unsized-locals/unsized-exprs.rs:22:26
   |
22 |     udrop::<(i32, [u8])>((42, *foo()));
   |                          ^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: within `({integer}, [u8])`, the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because it appears within the type `({integer}, [u8])`
   = note: tuples must have a statically known size to be initialized

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> src/test/ui/unsized-locals/unsized-exprs.rs:24:22
   |
24 |     udrop::<A<[u8]>>(A { 0: *foo() });
   |                      ^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: within `A<[u8]>`, the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because it appears within the type `A<[u8]>`
   = note: structs must have a statically known size to be initialized

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> src/test/ui/unsized-locals/unsized-exprs.rs:26:22
   |
26 |     udrop::<A<[u8]>>(A(*foo()));
   |                      ^ doesn't have a size known at compile-time
   |
   = help: within `A<[u8]>`, the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because it appears within the type `A<[u8]>`
   = note: the return type of a function must have a statically known size

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
```

Reply:
Ah I see, there's an unsized return value?

Seems a bit odd that you would need both feature gates, but maybe that makes sense?